experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -415,6 +415,9 @@ function shallowCompatibilityTest(
 			getFieldKind(primary.schema).multiplicity === Multiplicity.Sequence
 		);
 	}
+	if (data instanceof Map) {
```

Comment:
@Josmithr Adding this here allows the code that converts JSON into tree content to understand maps. Previously it said, "if data is a Map, I don't know what that is, so I'm just going to assume it's compatible with the schema". That fails in the case where you have a field that can either be a Map _or_ an Object, and you set it to be a new Map. The failure is because this code couldn't figure out which one you meant to insert. With this small change, it now recognizes Maps and will only say it matches the schema if the schema is a Map as well.